src/ln/channelmanager.rs

```diff
@@ -2622,6 +2622,20 @@ impl ChannelManager {
 
 impl events::MessageSendEventsProvider for ChannelManager {
 	fn get_and_clear_pending_msg_events(&self) -> Vec<events::MessageSendEvent> {
+		// TODO: Event release to users and serialization is currently race-y: its very easy for a
```

Comment:
Well any of the actual results are going to need to wait on an event clear anyway. eg if it results in sending an update_fulfill_htlc we still have to wait for them to call get_and_clear_pending_msg_events. It may even make sense to move the transaction broadcasting stuff into events as the whole broadcaster API is pretty awkward.

As for the race condition here - either we can serialize the pending events or we can just fail serialization if there are pending events and let the user go in a loop until there are none left, which I think is probably ok.

Reply:
Ah yes, so I think it's fine, acknowledging it's a best effort for now, need to spend a little bit more time with rust-lightning-bitcoinrpc to have more clues on how the broadcast/events APIs could be used